analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/UnexpectedDiagnosticException.cs

```diff
@@ -28,31 +28,31 @@ namespace SonarAnalyzer.UnitTest.TestFramework
     /// <summary>
     /// This exception class is used in unit tests that analyze some code to report a violation.
     /// It differs from classical unit test failure reporting in that it injects in its call
-    /// stack a location that correspond to the source file that is being analyzed (when analysing
+    /// stack a location that correspond to the source file that is being analyzed (when analyzing
     /// inline code snippets, this location will be incorrect).
     /// As a result, in the UI, we can just click on this line to directly open the right file at
     /// the right place to see the actual issue.
     /// </summary>
-    ///
-    public class UnexpectedDiagnosticException : Exception
+    public abstract class TestfileDiagnosticException : Exception
```

Comment:
Two derived classes for now:
* `UnexpectedDiagnosticException`
* `MissingDiagnosticException`

Maybe we can add `InvalidTestAnnotationException` for malformed annotations later.